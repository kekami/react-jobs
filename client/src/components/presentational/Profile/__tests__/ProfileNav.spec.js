import React from 'react';
import ReactDOM from 'react-dom';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MemoryRouter } from 'react-router-dom';
import { ProfileNav } from '../ProfileNav';

// Tests for Profile Navigation Section
/*
1. basic render
2. snapshot
3. one nav is rendered
4. two list items are rendered
*/

describe('ProfileNav - basic tests', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <ProfileNav />
      </MemoryRouter>, div);
  });

  it('ProfileNav matches its shallow snapshot', () => {
    const wrapper = shallow(<ProfileNav />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('ProfileNav should render one nav item', () => {
    const wrapper = shallow(<ProfileNav />);
    expect(wrapper.find('nav').length).toBe(1);
  });

  it('ProfileNav renders two navigation options', () => {
    const wrapper = shallow(<ProfileNav />);
    expect(wrapper.find('li').length).toBe(2);
  });
});
