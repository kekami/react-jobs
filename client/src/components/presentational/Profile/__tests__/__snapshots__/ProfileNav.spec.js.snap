// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileNav - basic tests ProfileNav matches its shallow snapshot 1`] = `
<styled.div>
  <nav>
    <ul>
      <li>
        <Link
          onClick={[Function]}
          replace={false}
          to="/profile"
        >
          <strong>
            MY PROFILE
          </strong>
        </Link>
      </li>
      <li>
        <Link
          onClick={[Function]}
          replace={false}
          to="/profile"
        >
          <strong>
            MY PREFERENCES
          </strong>
        </Link>
      </li>
    </ul>
  </nav>
</styled.div>
`;
